2024-01-25 19:50:54: Experiment log path in: D:\omer\ST-SSL\experiments\Careem\20240125-195054
2024-01-25 19:50:54: Experiment configs are: Namespace(seed=31, device='cuda', mode='train', best_path='None', debug=False, data_dir='data', dataset='Careem', input_length=19, batch_size=32, test_batch_size=1, graph_file='data/Careem/adj_mx.npz', d_input=1, d_output=1, d_model=64, dropout=0.2, percent=0.1, shm_temp=0.5, nmb_prototype=2, yita=1, epochs=1, lr_init=0.001, early_stop=True, early_stop_patience=150, grad_norm=True, max_grad_norm=5, use_dwa=True, temp=4, num_nodes=285, log_dir='D:\\omer\\ST-SSL\\experiments\\Careem\\20240125-195054')
2024-01-25 19:51:02: *******Train Epoch 1: averaged Loss : 9.391333
2024-01-25 19:51:03: Traceback (most recent call last):
  File "D:\omer\ST-SSL\main.py", line 59, in model_supervisor
    results = trainer.train() # best_eval_loss, best_epoch
              ^^^^^^^^^^^^^^^
  File "D:\omer\ST-SSL\model\trainer.py", line 115, in train
    val_epoch_loss = self.val_epoch(epoch, val_dataloader, loss_weights)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\omer\ST-SSL\model\trainer.py", line 84, in val_epoch
    loss, sep_loss = self.model.loss(repr1, repr2, target, self.scaler, loss_weights)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\omer\ST-SSL\model\models.py", line 66, in loss
    l2 = self.temporal_loss(z1, z2)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\omer\ST-SSL\model\models.py", line 83, in temporal_loss
    return self.thm(z1, z2)
           ^^^^^^^^^^^^^^^^
  File "C:\Users\CITY3\.conda\envs\i-jepaVENV\Lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\CITY3\.conda\envs\i-jepaVENV\Lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\omer\ST-SSL\model\layers.py", line 112, in forward
    logits = self.disc(s, h, shuf_h)
             ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\CITY3\.conda\envs\i-jepaVENV\Lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\CITY3\.conda\envs\i-jepaVENV\Lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\omer\ST-SSL\model\layers.py", line 157, in forward
    sc_fk = torch.squeeze(self.net(h_fk, s), dim=2)
                          ^^^^^^^^^^^^^^^^^
  File "C:\Users\CITY3\.conda\envs\i-jepaVENV\Lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\CITY3\.conda\envs\i-jepaVENV\Lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\CITY3\.conda\envs\i-jepaVENV\Lib\site-packages\torch\nn\modules\linear.py", line 198, in forward
    return F.bilinear(input1, input2, self.weight, self.bias)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: bilinear(): input batch dimensions do not match at dim 0: got 32 and 1

